app/static/src/app/userMessages.ts

```diff
@@ -25,6 +25,9 @@ export default {
     },
     modelFit: {
         cannotFit: "Cannot fit model. Please provide valid data and code, and link at least one variable.",
-        notFittedYet: "Model has not been fitted."
+        notFittedYet: "Model has not been fitted.",
+        compileRequired: "Model code has been updated. Compile code and Fit Model for updated best fit.",
+        fitRequired: "Model code has been recompiled, or options or data have been updated. "
```

Comment:
Is it possible to give more specific messages at this stage, so that the user doesn't need to figure out which condition they need to address?

Reply:
Yes, and actually that's even more of an issue with the next branch, which means they have to select at least one parameter to vary too. But I think this is better than nothing for now, so I'll make a new ticket to be more specific here, and just include those conditions which are not yet satisfied.

Reply:
https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-3470